Controller/Admin/ModuleConfigController.php

```diff
@@ -136,6 +141,46 @@ public function setRewritingEnableAction()
         }
     }
 
+    // check index_redirection_enable status & write an entry if there is none //
+    
+    public function setIndexRedirectionEnableAction()
+    {
+        $request = $this->getRequest()->request;
+        $isIndexRedirectionEnable = $request->get("index_redirection_enable", null);
+
+        if ($isIndexRedirectionEnable !== null) {
+            ConfigQuery::write("index_redirection_enable", $isIndexRedirectionEnable ? 1 : 0);
+            return $this->jsonResponse(json_encode(["state" => "Success"]), 200);
+        } else {
+            return $this->jsonResponse(Translator::getInstance()->trans(
+                "Unable to change the configuration variable.",
+                [],
+                RewriteUrl::MODULE_DOMAIN
+            ), 500);
```

Comment:
```suggestion
            ), 400);
```
C'est une erreur 400, car dans ton cas, si `$isIndexRedirectionEnable` est null, c'est que la requete n'est pas valide.  
https://developer.mozilla.org/fr/docs/Web/HTTP/Status/400  

Idem pour la méthode setHttpsRedirectionEnableAction.  

Tu peux aussi le corriger pour la méthode setRewritingEnableAction.

Et si tu veux même utiliser les bonnes pratiques, tu peux remplacer ce code
```php
return $this->jsonResponse(Translator::getInstance()->trans(
                "Unable to change the configuration variable.",
                [],
                RewriteUrl::MODULE_DOMAIN
            ), 500);
```
par
```
throw new BadRequestHttpException('Missing index_redirection_enable parameter in url');
```
Le Kernel symfony va intercepter cette exception pour créer une réponse 400 avec le message.  
Tu peux aussi réorganiser ce code pour virer le `else`.  
Et pas besoin de traduire cette erreur, en théorie elle n'est pas visible pour un simple utilisateur.  Et quand l'erreur est destinée à des développeurs, l'anglais est largement suffisant.